resources/js/Hooks/useAuth.ts

```diff
@@ -43,12 +74,26 @@ export const useAuth = (): App.Data.AuthData & {
         }
 
         return requiresSignature;
-    }, [authenticated, connecting, metamaskErrorMessage, requiresSignature, switching, error]);
+    }, [
+        authenticated,
+        connecting,
+        metamaskErrorMessage,
+        requiresSignature,
+        switching,
+        error,
+        allowsGuests,
+        manuallyClosed,
+        isShowConnectOverlay,
+    ]);
+
+    const showCloseButton = allowsGuests || isShowConnectOverlay;
```

Comment:
what is `isShowConnectOverlay` added for here? this causes it to show the button also on pages where it's not allowed to view it as a guest 🤔